src/HttpClientPool/RoundRobinClientPool.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Http\Client\Common\HttpClientPool;
+
+use Http\Client\Common\Exception\HttpClientNotFoundException;
+use Http\Client\Common\HttpClientPool;
+
+/**
+ * RoundRobinClientPool will choose the next client in the pool.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class RoundRobinClientPool extends HttpClientPool
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function chooseHttpClient()
+    {
+        $last = current($this->clientPool);
```

Comment:
We can maybe add a third strategy which randomly choose a client between the one that are available ?